src/guacscanner/guacscanner.py

```diff
@@ -0,0 +1,764 @@
+                error="Possible values for --log-level are "
+                + "debug, info, warning, error, and critical.",
+            ),
+            "--sleep": And(
+                Use(float),
+                error="Value for --sleep must be parseable as a floating point number.",
+            ),
+            Optional("--vpc-id"): Or(
+                None,
+                And(
+                    str,
+                    Use(str.lower),
+                    lambda x: VPC_ID_REGEX.match(x) is not None,
+                    error="Possible values for --vpc-id are the characters vpc- followed by either 8 or 17 hexadecimal digits.",
+                ),
+            ),
+            str: object,  # Don't care about other keys, if any
+        }
+    )
+    try:
+        validated_args = schema.validate(args)
+    except SchemaError as err:
+        # Exit because one or more of the arguments were invalid
+        print(err, file=sys.stderr)
+        sys.exit(1)
+
+    # Set up logging
+    log_level = validated_args["--log-level"]
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
+    )
+
+    add_instance_states = DEFAULT_ADD_INSTANCE_STATES
+    postgres_db_name = DEFAULT_POSTGRES_DB_NAME
+    postgres_hostname = DEFAULT_POSTGRES_HOSTNAME
+    postgres_port = DEFAULT_POSTGRES_PORT
+    remove_instance_states = DEFAULT_REMOVE_INSTANCE_STATES
+
+    oneshot = validated_args["--oneshot"]
+    logging.debug("oneshot is %s.", oneshot)
+
+    postgres_password = validated_args["--postgres-password"]
+    if postgres_password is None:
+        with open(validated_args["--postgres-password-file"], "r") as file:
+            postgres_password = file.read()
+
+    postgres_username = validated_args["--postgres-username"]
+    if postgres_username is None:
+        with open(validated_args["--postgres-username-file"], "r") as file:
+            postgres_username = file.read()
+
+    rdp_password = validated_args["--rdp-password"]
+    if rdp_password is None:
+        with open(validated_args["--rdp-password-file"], "r") as file:
+            rdp_password = file.read()
+
+    rdp_username = validated_args["--rdp-username"]
+    if rdp_username is None:
+        with open(validated_args["--rdp-username-file"], "r") as file:
+            rdp_username = file.read()
+
+    vnc_password = validated_args["--vnc-password"]
+    if vnc_password is None:
+        with open(validated_args["--vnc-password-file"], "r") as file:
+            vnc_password = file.read()
+
+    vnc_username = validated_args["--vnc-username"]
+    if vnc_username is None:
+        with open(validated_args["--vnc-username-file"], "r") as file:
+            vnc_username = file.read()
+
+    private_ssh_key = validated_args["--private-ssh-key"]
+    if private_ssh_key is None:
+        with open(validated_args["--private-ssh-key-file"], "r") as file:
+            private_ssh_key = file.read()
```

Comment:
You shouldn't get the usage output in this case.  If you don't define either parameter explicitly then you get the default value for the `--postgres-password-file` option, which is `/run/secrets/postgres-password`.  If that file does not exist, then you get the error you showed in your comment.

I don't _think_ any changes are necessary here, but I'm open to being convinced otherwise.

Reply:
Ah, you're right. I missed the defaults in the giant wall of text. I suppose #5 would resolve the not-very-helpful output if you neglect to give an argument and it bombs out like this.

Reply:
Yes